File: backend/src/models/Consultation.js
const mongoose = require('mongoose');

const consultationSchema = new mongoose.Schema(
  {
    patient: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Patient',
      required: true,
    },
    clinique: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Clinique',
      required: true,
    },
    medecin: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    date: { type: Date, default: Date.now },
    motif: { type: String, required: true },
    diagnostic: { type: String, required: true },
    notes: { type: String, default: '' },
    constantes: {
      tension: { type: String, default: '' },
      temperature: { type: Number, default: null },
      poids: { type: Number, default: null },
      frequenceCardiaque: { type: Number, default: null },
    },
    createdAt: { type: Date, default: Date.now },
  },
  { timestamps: true }
);

const Consultation = mongoose.model('Consultation', consultationSchema);

module.exports = Consultation;

File: lib/models/consultation.dart
class Constantes {
  final String tension;
  final double? temperature;
  final double? poids;
  final int? frequenceCardiaque;

  Constantes({
    required this.tension,
    this.temperature,
    this.poids,
    this.frequenceCardiaque,
  });

  factory Constantes.fromJson(Map<String, dynamic> json) {
    return Constantes(
      tension: json['tension'] ?? '',
      temperature: (json['temperature'] as num?)?.toDouble(),
      poids: (json['poids'] as num?)?.toDouble(),
      frequenceCardiaque: json['frequenceCardiaque'] as int?,
    );
  }
}

class Consultation {
  final String id;
  final String patientId;
  final String motif;
  final String diagnostic;
  final String notes;
  final Constantes constantes;
  final String medecinNom;
  final String cliniqueNom;
  final DateTime date;

  Consultation({
    required this.id,
    required this.patientId,
    required this.motif,
    required this.diagnostic,
    required this.notes,
    required this.constantes,
    required this.medecinNom,
    required this.cliniqueNom,
    required this.date,
  });

  factory Consultation.fromJson(Map<String, dynamic> json) {
    return Consultation(
      id: json['_id'] ?? '',
      patientId: json['patient'] ?? '',
      motif: json['motif'] ?? '',
      diagnostic: json['diagnostic'] ?? '',
      notes: json['notes'] ?? '',
      constantes: Constantes.fromJson(json['constantes'] ?? {}),
      medecinNom: json['medecinNom'] ?? 'N/A',
      cliniqueNom: json['cliniqueNom'] ?? 'N/A',
      date: DateTime.parse(json['date'] ?? DateTime.now().toIso8601String()),
    );
  }
}
